modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/shp/PointHandler.java

```diff
@@ -78,22 +79,28 @@ public Object read(ByteBuffer buffer, ShapeType type, boolean flatGeometry) {
             return createNull();
         }
 
-        int dimension = shapeType == ShapeType.POINTZ && !flatGeometry ? 3 : 2;
-        CoordinateSequence cs =
-                JTS.createCS(geometryFactory.getCoordinateSequenceFactory(), 1, dimension);
+        Coordinate c;
+        if (shapeType == ShapeType.POINTZ) {
 
-        cs.setOrdinate(0, 0, buffer.getDouble());
-        cs.setOrdinate(0, 1, buffer.getDouble());
+            c = new CoordinateXYZM();
```

Comment:
I've dropped the CoordinateSequence here as it didn't seem to add much as only one coordinate was used. Or am I missing some subtle speedup that could occur?